scripts/zones/Davoi/npcs/Lootblox.lua

```diff
@@ -123,22 +147,37 @@ function onEventUpdate(player, csid, option)
 end
 
 function onEventFinish(player, csid, option)
+    local introMask = player:getCharVar("currencyGoblinIntro")
+
+    if csid == 131 then
+        -- make prismatic hourglass introduction unavailable to the other goblins
+        player:setCharVar("currencyGoblinIntro", utils.mask.setBit(introMask, 4, true))
+
+    elseif csid == 132 then
+        -- remove if this was the last of the three goblins that introduced currency
+        if utils.mask.getBit(introMask, 2) and utils.mask.getBit(introMask, 3) then
```

Comment:
can just check if the mask is full right after setting this goblin's bit, theres a function for it along with the get's for the bits.

Reply:
it works either way I just want you to know that exists. I was looking for it and missed your zero'ing out the var at 1st.

Reply:
I very much like this pointer and implemented it. It's a cleaner check and less error prone.